typing/includemod_errorprinter.ml

```diff
@@ -518,7 +520,10 @@ module Functor_suberror = struct
       | Named _ | Anonymous ->
           Format.dprintf
             "The functor was expected to be generative at this position"
-
+      | Empty_struct ->
+          (* an empty structure can be used in both applicative and generative
+             context *)
+          assert false
```

Comment:
I'm always nervous about new `assert false` in the code, but it looks widely used in this file for understandable reason, so this one is arguably okay.